packages/analytics-browser/src/browser-client.ts

```diff
@@ -116,7 +120,11 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
       this.q.push(this.setUserId.bind(this, userId));
       return;
     }
-    this.config.userId = userId;
+    if (userId !== this.config.userId) {
+      this.previousSessionUserId = this.config.userId;
+      this.config.userId = userId;
+      this.setSessionId(Date.now());
```

Comment:
do you think we need an option for this?

Reply:
We can but i was thinking that having a different user id make them not share sessions anyway right?

Reply:
You mean they will appear on different users? Make sense.